packages/trie/src/trie/trie.ts

```diff
@@ -176,9 +194,22 @@ export class Trie {
   async del(key: Buffer): Promise<void> {
     await this.lock.wait()
     const { node, stack } = await this.findPath(key)
+    let ops: BatchDBOp[]
+    if (this._pruneTrie) {
+      const deleteHashes = stack.map((e) => this.hash(e.serialize()))
```

Comment:
Ah I just realized an edge case; do NOT delete keys if the value is the current value at `key`